### PR TITLE
Fixes #24193 - ensures better multipexception handling in core

### PR DESCRIPTION
**Pull request by rabajaj0509**

### CODE REVIEW

File: lib/foreman/exception.rb [hunk 1]
```diff
@@ -55,6 +55,28 @@ def message
...
+      raise "Can't create multi exception - not an array: #{exceptions}" unless exceptions.respond_to?(:to_a)
+      @exceptions = exceptions.to_a.dup.freeze
+    end
+
+    attr_reader :exceptions
+
+    def string_form
+      super if @wrapped_exception.empty?
+      super + @exceptions.map { |e| e.message }
+    end
```
Comment (lzap) [inline lib/foreman/exception.rb:74]: Please override `to_s` from Exception class instead new method name.

File: lib/foreman/exception.rb [hunk 2]
```diff
@@ -55,6 +55,28 @@ def message
...
+
+    def json_form
+      {
+        message: message,
+        errors: (@exceptions.map { |e| e.message })
+      }.as_json
+    end
```
Comment (lzap) [inline lib/foreman/exception.rb:79]: Please create `as_json` method in Exception base class and override here (call it and merge `errors` key into the hash).

Also I suggest to do something like: `e.respond_to?(:message) ? e.message : e.to_s`

**Review by lzap — CHANGES_REQUESTED** [review]
Looks good, this will be useful class.

**Comment by lzap** [comment]
What is the status here?

**Comment by rabajaj0509** [comment]
@lzap looks good?

**Review by lzap — APPROVED** [review]
Ok I will rewrite the message on push.

**Pull request merged**